tests/BreadcrumbTest.php

```diff
@@ -43,6 +43,39 @@ public function testConstructor(): void
         $this->assertEquals(microtime(true), $breadcrumb->getTimestamp());
     }
 
+    public function testFromArray(): void
+    {
+        $breadcrumb = Breadcrumb::fromArray([
+            'level' => Breadcrumb::LEVEL_INFO,
+            'type' => Breadcrumb::TYPE_USER,
+            'category' => 'foo',
+            'message' => 'foo bar',
+            'data' => ['baz'],
+        ]);
+
+        $this->assertEquals('foo', $breadcrumb->getCategory());
+        $this->assertEquals(Breadcrumb::LEVEL_INFO, $breadcrumb->getLevel());
+        $this->assertEquals('foo bar', $breadcrumb->getMessage());
+        $this->assertEquals(Breadcrumb::TYPE_USER, $breadcrumb->getType());
+        $this->assertEquals(['baz'], $breadcrumb->getMetadata());
+        $this->assertEquals(microtime(true), $breadcrumb->getTimestamp());
+    }
+
+    public function testFromArrayDefaults(): void
```

Comment:
Instad of splitting the test into two different methods, simply use the one above with a data provider that provides the input `$data` parameter to pass to the `fromArray` method and an `$expectedResult` that can be compared to what the `toArray` method returns

Reply:
ok.  I think I've done as you ask now.  please review.